### ISSUE TITLE
Adding Pulumi as Manager

### ISSUE THREAD

**Issue by piclemx** [renovatebot/renovate#5794]
**What would you like Renovate to be able to do?**
Pulumi is a tool like Terraform, Helm, Ansible and others that do the infrastructure as code. It will be great add to the project.

**Describe the solution you'd like**
I will be great to detect the container image and updated it if possible. Otherwise since it used real language. Javascript,Golang,Python are already handle.

**Describe alternatives you've considered**
None

**Additional context**

```
import * as kx from "@pulumi/kubernetesx";

const pb = new kx.PodBuilder({
    containers: [{
        image: "nginx",
        ports: { http: 80 }
    }]
});

const deployment = new kx.Deployment("nginx", {
    spec: pb.asDeploymentSpec({ replicas: 3 })
});

const service = deployment.createService({
    type: kx.types.ServiceType.LoadBalancer
});

export const serviceIP = service.ip;
```

**Comment by rarkins** [comment]
Does Pulumi have any concept of linking to its own datasources, e.g. like Helm charts link to Helm charts? Or is it specifying containers, golang packages, etc?

**Comment by viceice** [comment]
if you only need to update `image: "nginx:2.3.45"`, you can use the regex manager

**Comment by piclemx** [comment]
@rarkins It's using containers image for k8s.

**Comment by viceice** [comment]
https://www.pulumi.com/docs/get-started/azure/install-language-runtime/

Looks like we can't create a simple manager for this, cause you write your  pulumi files in any language (typescript /javascript /python /go /c# ...). So it wouild need language agnostic parsers to find the right things to update.

Maybe for now the easiest solution would be to write the deps to a key-value file (eg ini) add renovate comments and a regex manager to update this file. and that read the deps from pupuli code instead of inlining the deps

**Comment by elsbrock** [comment]
Provider dependencies should already be managed I assume, since they are usually specified as  
package dependency of your project.

**Comment by piclemx** [comment]
@elsbrock  Yes, for the provider coming from multiple languages. You can't do that with the version of a container.

**Comment by simenandre** [comment]
Hello 👋 

I suggest using Pulumi stack files. These are YAML files, which are used to configure things. Using these YAML files to define container tags is considered a good practice. We might even introduce a standardized way of defining tags.

A real-world example:
https://github.com/getbranches/conf/blob/main/Pulumi.main.yaml

Notice Pulumi._main_.yaml. In the example, this stack is called _main_. Typically, there's more than one of these files, one for each stack/environment (i.e. `Pulumi.dev.yaml` and `Pulumi.prod.yaml`). In my example, I only have one.

**Comment by simenandre** [comment]
Hello!

For reference, I have experimented with using `regexManager` for this. It works fine, for what I'm doing.

See https://github.com/getbranches/conf
